proxy/proxy.go

```diff
@@ -1153,6 +1167,7 @@ func (p *Proxy) serveResponse(ctx *context) {
 	copyHeader(ctx.responseWriter.Header(), ctx.response.Header)
 	p.tracing.logStreamEvent(ctx.proxySpan, StreamHeadersEvent, EndEvent)
 
+	// TODO differentiate client cancelation and backend timeout before streaming
```

Comment:
Needs to be implemented and test covered, see similar https://github.com/zalando/skipper/pull/1566#discussion_r538839840

Reply:
Added the test for client cancellation. This codepath is not a subject of backendTimeout as it configures timeout on backend request, not on client request.